datasource/src/main/scala/quasar/physical/mongo/Mongo.scala

```diff
@@ -114,19 +132,22 @@ class Mongo[F[_]: MonadResourceErr : ConcurrentEffect] private[Mongo](client: Mo
   private val maximumBatchBytes: Long = if (maxMemory > MaxBsonBatch) MaxBsonBatch else maxMemory.toLong
 
   def getQueueSize(collection: Collection): F[Long] = {
-    val getStats: F[Document] = F.suspend(singleObservableAsF[F, Document](
+    val getStats: F[Option[Document]] = F.attempt(F.suspend(singleObservableAsF[F, Document](
       client.getDatabase(collection.database.name).runCommand[Document](Document(
         "collStats" -> collection.name,
         "scale" -> 1
       ))
-    ))
+    ))).map(_ match {
+      case Left(e) => None
+      case Right(a) => Some(a)
+    })
 
     getStats map { doc =>
-      val optSize: Option[Long] = doc.get("avgObjSize").flatMap(_ match {
+      val optSize: Option[Long] = doc flatMap (_.get("avgObjSize") flatMap (_ match {
```

Comment:
Stylistically, I would prefer:

```scala
doc.flatMap(_.get("avgObjSize")) flatMap {
```

Remember that monads are associative, so there's no need to nest the binds here.